### PR TITLE
add solution

### PR DESCRIPTION
**Pull request by AndrewHudzenko**

### CODE REVIEW

File: src/main/java/mate/academy/dao/impl/ShoppingCartDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<ShoppingCart> getByUser(User user) {
+        return Optional.empty();
```
Comment (Nesterenko-Andrey) [inline src/main/java/mate/academy/dao/impl/ShoppingCartDaoImpl.java:40]: I think we should have the realisation here

File: src/main/java/mate/academy/dao/impl/ShoppingCartDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    @Override
+    public void update(ShoppingCart shoppingCart) {
+
```
Comment (Nesterenko-Andrey) [inline src/main/java/mate/academy/dao/impl/ShoppingCartDaoImpl.java:45]: same here

**Review by Nesterenko-Andrey — CHANGES_REQUESTED** [review]

**Review by Nesterenko-Andrey — APPROVED** [review]

File: src/main/java/mate/academy/Main.java [hunk 3]
```diff
@@ -44,12 +54,29 @@ public static void main(String[] args) {
...
+        try {
+            user = authenticationService.register("bob@gmail.com", "12345678");
+        } catch (RegistrationException e) {
+            throw new RuntimeException("Couldn't register", e);
+        }
+
+        ShoppingCartService shoppingCartService =
+                (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+        shoppingCartService.addSession(yesterdayMovieSession, user);
+        shoppingCartService.registerNewShoppingCart(user);
```
Comment (Sofasmile) [inline src/main/java/mate/academy/Main.java:78]: should you register a shopping cart here?

File: src/main/java/mate/academy/dao/impl/ShoppingCartDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.User;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+
```
Comment (Sofasmile) [inline src/main/java/mate/academy/dao/impl/ShoppingCartDaoImpl.java:16]: remove empty line

File: src/main/java/mate/academy/security/AuthenticationServiceImpl.java [hunk 5]
```diff
@@ -30,6 +33,7 @@ public User register(String email, String password) throws RegistrationException
             user.setEmail(email);
             user.setPassword(password);
             userService.add(user);
+            shoppingCartService.registerNewShoppingCart(user);
```
Comment (Sofasmile) [inline src/main/java/mate/academy/security/AuthenticationServiceImpl.java:36]: 👍

**Review by Sofasmile — CHANGES_REQUESTED** [review]

**Pull request closed without merging**